lib/IRGen/GenPack.cpp

```diff
@@ -454,6 +454,8 @@ irgen::emitTypeMetadataPack(IRGenFunction &IGF,
   auto pack = IGF.emitDynamicAlloca(IGF.IGM.TypeMetadataPtrTy, shape,
                                     IGF.IGM.getPointerAlignment(),
                                     /*allowTaskAlloc=*/true);
+  pack = pack.withAddress(IGF.Builder.CreateElementBitCast(
```

Comment:
Should we do this inside emitDynamicAlloca() or is it not worth refactoring since we're switching to opaque pointers at some point?

Reply:
Yeah, let's do that.